addons/marketing_card/models/card_card.py

```diff
@@ -8,14 +8,15 @@ class MarketingCard(models.Model):
     _name = 'card.card'
     _description = 'Marketing Card'
 
+    active = fields.Boolean('Active', default=True)
     campaign_id = fields.Many2one('card.campaign', required=True, ondelete="cascade")
-    res_model = fields.Selection(related='campaign_id.res_model')
-    record_ref = fields.Reference(string='Record', selection='_selection_record_ref', compute='_compute_record_ref')
+    res_model = fields.Char(related='campaign_id.res_model')
     res_id = fields.Many2oneReference('Record ID', model_field='res_model', required=True)
     image = fields.Image()
+    generation = fields.Integer(help="If matching the generation of the campaign, is up to date.")
```

Comment:
Ok, it is used as a way to detect outdated cards ... but it seems like a weird implementation to me. Can't we have an "outdated", "dirty", ... flag on cards ? Changing any "render related" field on campaign would write "dirty" on all its cards. Then we know those have to be generated again. Comparing against "version number" is weird, I would prefer having a real flag on cards.